packages/ethereum/src/Network/Ethereum/Contract/TH.hs

```diff
@@ -120,7 +120,8 @@ toHSType s = case s of
     SolidityString      -> conT ''Text
     SolidityBytesN n    -> appT (conT ''BytesN) (numLit n)
     SolidityBytes       -> conT ''Bytes
-    SolidityTuple n as  -> foldl ( \b a -> appT b $ toHSType a ) ( tupleT n ) as
```

Comment:
Having the length part of the structure made the new case awkward due to impossible cases on pattern match, so I changed it to be computed from the list itself.